backend/common/src/ws_client/connect.rs

```diff
@@ -228,11 +236,18 @@ pub enum ConnectError {
 /// Establish a websocket connection that you can send and receive messages from.
 pub async fn connect(uri: &http::Uri) -> Result<Connection, ConnectError> {
     let host = uri.host().unwrap_or("127.0.0.1");
-    let port = uri.port_u16().unwrap_or(80);
+    let scheme = uri.scheme_str().unwrap_or("ws");
+    let mut port = 80;
+    if scheme == "https" || scheme == "wss" {
+        port = 443
+    }
     let path = uri.path();
-
+    let port = uri.port_u16().unwrap_or(port);
     let socket = TcpStream::connect((host, port)).await?;
     socket.set_nodelay(true).expect("socket set_nodelay failed");
+    // wrap TCP stream with TLS if schema is https or wss
+    let socket =
+        may_connect_tls(socket, host.clone(), scheme == "https" || scheme == "wss").await?;
```

Comment:
this clone seems needless, it's a &str right?

```suggestion
        may_connect_tls(socket, host, scheme == "https" || scheme == "wss").await?;
```

Reply:
Ooh good catch!